cda-comm-doip/src/ecu_connection.rs

```diff
@@ -57,12 +60,12 @@ pub(crate) struct EcuConnectionTarget {
 }
 
 impl ECUConnection for EcuConnectionVariant {
-    async fn send(&mut self, msg: DoipPayload) -> Result<(), String> {
+    async fn send(&mut self, msg: DoipPayload) -> Result<(), DiagServiceError> {
         match self {
             EcuConnectionVariant::Tls(conn) => conn.send(msg).await,
             EcuConnectionVariant::Plain(conn) => conn.send(msg).await,
         }
-        .map_err(|e| format!("Failed to send message: {e:?}"))
+        .map_err(|e| DiagServiceError::SendFailed(format!("Failed to send message: {e:?}")))
```

Comment:
question: What's the reason for using DiagServiceError instead of ConnectionError?

Reply:
Actually that was a genuine question, but using ConnectionError now is fine too 👍